### PR TITLE
Fix Examples to work with 1.1-beta2

### PR DESCRIPTION
**Pull request by anton-bannykh**

### CODE REVIEW

**Comment by anton-bannykh** [comment]
ping?

File: kotlin.web.demo.server/examples/Examples/Canvas/Creatures/Creatures.kt [hunk 1]
```diff
@@ -208,13 +208,17 @@ class CanvasState(val canvas: HTMLCanvasElement) {
...
     var valid = false
-    var shapes = ArrayList<Shape>()
+    var shapes = mutableListOf<Shape>()
     var selection: Shape? = null
     var dragOff = Vector()
     val interval = 1000 / 30
 
     init {
-        jq(canvas).mousedown {
+        fun fix(f: Element.(MouseEvent) -> Unit) : Element.(MouseEvent) -> Unit {
```
Comment (yole) [inline kotlin.web.demo.server/examples/Examples/Canvas/Creatures/Creatures.kt]: What's this about and why is this needed? Can we avoid showing such ugly stuff in samples?

Comment (anton-bannykh) [inline kotlin.web.demo.server/examples/Examples/Canvas/Creatures/Creatures.kt]: AFAIK there is no way to avoid hacks at the moment. cc @bashor 

The way extension lambdas are translated has been changed. They used to get the receiver parameter from `this`. Now there is a dedicated parameter `$receiver` for that. That was done in order to use extension and regular lambdas interchangeably (e.g. so that (A).(B) -> C is equivalent to (A, B) -> C).

Comment (bashor) [inline kotlin.web.demo.server/examples/Examples/Canvas/Creatures/Creatures.kt]: that's right

**Comment by ilya-g** [comment]
merging now, `fix` for jquery can be polished later

File: kotlin.web.demo.server/examples/Examples/Canvas/Creatures/Creatures.kt (hunk 1 above)
Comment (ilya-g) [inline kotlin.web.demo.server/examples/Examples/Canvas/Creatures/Creatures.kt]: So do we need some special extension lambda to be passed to jquery event handler?

Comment (bashor) [inline kotlin.web.demo.server/examples/Examples/Canvas/Creatures/Creatures.kt]: Yes, it will be designed separately

Comment (anton-bannykh) [inline kotlin.web.demo.server/examples/Examples/Canvas/Creatures/Creatures.kt]: jQuery event handler expects a function `foo`, which it then uses like this: `f.call(element, mousevent)`. Currently there is no way to create a lambda that could be used like that.
